api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -32,19 +34,13 @@
  * @see org.openmrs.api.context.Context
  * @see org.openmrs.api.CohortService
  */
+@Repository("cohortDAO")
 public class HibernateCohortDAO implements CohortDAO {
 	
 	private static final String VOIDED = "voided";
-	private SessionFactory sessionFactory;
 	
-	/**
-	 * Auto generated method comment
-	 *
-	 * @param sessionFactory
-	 */
-	public void setSessionFactory(SessionFactory sessionFactory) {
-		this.sessionFactory = sessionFactory;
-	}
+	@Autowired
+	private SessionFactory sessionFactory;
```

Comment:
I should have mentioned it on the issue, but we want to use constructor autowiring instead of fields.